### PR TITLE
typo corrections

### PR DESCRIPTION
**Pull request by zio-pietro**
adjusted pillar links at multiple lines
adjusted == == bad rendering
added some LaTeX suggar

### CODE REVIEW

File: Preface/Preface.pillar [hunk 1]
```diff
@@ -58,12 +57,13 @@ available at *http://www.deepintopharo.org*
 
 Do not be frustrated by parts of Pharo that you do not immediately understand.
 You do not have to know everything! Alan Knight expresses this as follows:
-
+{{{latex:\begin{quote} }}}
```
Comment (DamienCassou) [inline Preface/Preface.pillar]: Use @@note instead. For example

```
@@note ""Try not to care"". ...
```

This generates a begin/end block in LaTeX and a div in HTML.

File: Preface/Preface.pillar [hunk 2]
```diff
@@ -83,16 +83,14 @@ that you may find useful:
...
-and projects for Pharo live there.
+- SmalltalkHub (*http://www.smalltalkhub.com/>http://www.smalltalkhub.com/*) is the equivalent of SourceForge/Github for Pharo projects. Many extra packages and projects for Pharo live there.
 
 !!Examples and exercises
 
 We have tried to provide as many examples as possible. In particular, there are
 many examples that show a fragment of code which can be evaluated. We use the
-symbol ==-->== an expression and to indicate the result that you obtain when you
-select an expression and ""print it"":
+symbol {{{latex:$\to$}}}{{{html:<code>--&gt;</code>}}} to indicate the result
```
Comment (DamienCassou) [inline Preface/Preface.pillar]: We must keep `-->` everywhere. Yes, it is less nice, but this is what will appear inside code blocks

**Comment by DamienCassou** [comment]
Thank you. But please take care of my comments.

/cc @kilon @zio-pietro

File: Preface/Preface.pillar [hunk 3]
```diff
@@ -83,16 +83,14 @@ that you may find useful:
...
-- SmalltalkHub (*http://www.smalltalkhub.com/>http://www.smalltalkhub.com/*)
-is the equivalent of SourceForge/Github for Pharo projects. Many extra packages
-and projects for Pharo live there.
+- SmalltalkHub (*http://www.smalltalkhub.com/>http://www.smalltalkhub.com/*) is the equivalent of SourceForge/Github for Pharo projects. Many extra packages and projects for Pharo live there.
 
 !!Examples and exercises
 
 We have tried to provide as many examples as possible. In particular, there are
 many examples that show a fragment of code which can be evaluated. We use the
-symbol ==-->== an expression and to indicate the result that you obtain when you
```
Comment (DamienCassou) [inline Preface/Preface.pillar]: To write `--` in Pillar, you have to escape with a backslash (because `--` means strikethrough):

```
- symbol ==-\->==
```

**Comment by kilon** [comment]
done with 
https://github.com/SquareBracketAssociates/UpdatedPharoByExample/commit/655bbca09d513dd88ca927b60c438c11366cfcb5

**Comment by DamienCassou** [comment]
Thanks @kilon, you are doing a great job!

**Comment by kilon** [comment]
I have a good teacher ;)

File: Preface/Preface.pillar (hunk 3 above)
Comment (zio-pietro) [inline Preface/Preface.pillar]: Hi Damien,
there is IMHO a little confusion. I'm not author of this text, I'm proofreader. (and Pillar newbie, too)
... in Pillar, you have to escape with a backslash ... yes, that's why in 30 pages of Pillar documentation
backslash as escape character isn't even mentioned ;)

File: Preface/Preface.pillar (hunk 1 above)
Comment (dmitrizagidulin) [inline Preface/Preface.pillar]: @DamienCassou - I would actually argue with the recommendation of ``@@note`` instead of a LaTeX quote tag (``{{{latex:\begin{quote} }}}``).

Pillar ``@@note`` prepends **Notes:** to the section, when rendering it in PDF. There are two problems with this: 1) It says Notes plural. Which is incorrect semantically -- the line is a single Note. 2) More importantly, this is NOT a note, but an actual quote from an author. And so should be formatted differently.

File: Preface/Preface.pillar (hunk 2 above)
Comment (zio-pietro) [inline Preface/Preface.pillar]: I did for now only 1st chapter corrections. If this change will be accepted, it's obvious I'll change it in whole book.
We must keep ugly rendering because we must? We are doing *book*, not 3 pages article. (Yes, I'm prefectionist ;)
...  but this is what will appear inside code blocks ... not so right, nay, it's not right at all. See http://imgbin.org/index.php?page=image&id=26981 symbol used by me is identical to that used in code block and I didn't touched source of code block or it's rendering.

File: Preface/Preface.pillar (hunk 1 above)
Comment (zio-pietro) [inline Preface/Preface.pillar]: Yes, I tried it, with unsatisfactory result. It'isn't note, as rendered Notes: Try not to care. ... it's citation, quote of Alan Knight. Thus I used LaTeX quote ambient, so appropriate for this type of text. 
Btw. Kilon and me have (not so) small talk on #pharo about Pillar as tool for writing book. I consider Pillar good tool for writing short texts, but for book? My personal opinion: a reasonable person can recognize when he made a mistake and can change his decision. IMHO decision using Pillar for writing book was one of that mistakes. When we will change this decision, I am ready to help as I can. I'm ready to help even if we will not change that decision,
with gritted teeth I will fight with flaws of Pillar to produce the best book as I can ;) (I'm not expert, but rather well skilled LaTeX user for 15+ years)

Comment (dmitrizagidulin) [inline Preface/Preface.pillar]: @zio-pietro - I changed it back to use the latex quote and an html blockquote (https://github.com/SquareBracketAssociates/UpdatedPharoByExample/commit/828f7603bf030bc2c2576e9490cbb2e3cb74a638 and https://github.com/SquareBracketAssociates/UpdatedPharoByExample/commit/3390f2c3260b781fca1b4aa7bfba9f44e5abcd95). Pillar really needs a dedicated Quote syntax (a multi-line one, either based on indentation like Markdown, or at least with an open/close syntax like Pillar code blocks).

As far as "using Pillar for book writing is a mistake" - I promise you, this is a losing battle, for many reasons :)
Politically, the Powers That Be (the Pharo association, and the actual writers of these books) are very committed to using Pillar instead of, say, Markdown. They are highly unlikely to change their minds on this. (There are several discussions / heated arguments on the Pharo mailing list, on this subject).

But practically, what do you think the book should be written in, instead? Aside from Pillar, there are basically only 3 options:

1) LaTeX - nope. I know you're a skilled LaTeX user, but this is not a realistic format, in terms of crowd-sourced book writing. Aside from a very small handful of CompSci academic types, LaTeX is a nightmare for regular people to use. Also, note that the first edition of Pharo By Example was actually written in LaTeX (there's still a repo for it, linked off the README). And people went through *considerable* effort to rewrite it from LaTeX to Pillar.

2) Markdown (with conversion to HTML and PDF via something like [Pandoc](http://pandoc.org/)).

3) Asciidoc (with conversion to HTML and PDF via [Asciidoctor](http://asciidoctor.org) plugins). 

In terms of options 2 & 3 (markdown and asciidoc).. they're basically equivalent to Pillar. Yes, a bit more people know Markdown (due to Github, etc) than do Pillar. (I would argue that Asciidoc is equally obscure). The capabilities are largely the same, and they're all equally easy to learn. 

Yes, Pillar does have some limitations when compared to Markdown (lack of indentation-based multi-line parsing, for example, for things like Lists, etc). But the surprising thing is, Pillar is moving and evolving way faster than Markdown is. 

So, really, the Pharo community's insistence on using Pillar is deep down about *controlling the means of production*. While Pillar is being actively maintained, we can make feature requests and bug reports (actually, I think @DamienCassou is one of the contributors to Pillar). 

To sum up, the move from Pillar to something else (like Markdown) ain't gonna happen, because a) people have *really* strong opinions in its favor, b) it's essentially equivalent to Markdown anyway, and c) Oh my god, it took lots of man-years of effort to convert the book from LaTeX to Pillar, why would anybody want to convert it to Markdown again? (Although I heard that Pillar renders to Markdown, actually, so never mind that part).

Comment (zio-pietro) [inline Preface/Preface.pillar]: @dmitrizagidulin  yes, I'm slowly discovering the background (personally I think it's a kind of NIH syndrom), but I'll do my best of helping us to do good UPBE from typographics point of view even with Pillar. Still remain of opinion for short texts any markup language, for big works LaTeX. "Here I stand. I can do no other"
[two hours long continuation deleted, I know, it's a lost battle, today's people want quick results with minimal effort and so-so quality is good enough]
(I know lawyer, who convert all his docs even to pure TeX for best results :) )

Comment (DamienCassou) [inline Preface/Preface.pillar]: @dmitrizagidulin When you prefix a line with `@@XXX` Pillar generates a LaTeX environment named `XXX`. It's up to the book author to define this `XXX` LaTeX environment. This is typically in a file named `support/latex/common.tex`.

Use `@@quote` when it's a quote. `@@note` should generate a paragraph with "Note" and not "Notes". Please fix it.

Comment (DamienCassou) [inline Preface/Preface.pillar]: I also like LaTeX. Our previous books were written in LaTeX but, we changed because:

1. LaTeX can't generate nice HTML
2. LaTeX is heavy to install so people tend not to contribute
3. LaTeX has much more features which makes it a bit harder to learn

Comment (DamienCassou) [inline Preface/Preface.pillar]: @dmitrizagidulin Yes, I'm the main contributor of Pillar

Pillar contains many features that are not in Markdown (e.g., references to sections).

Comment (kilon) [inline Preface/Preface.pillar]: I love Pillar and I dont like the verbosity of Latex, it reminds me too much of html/xml. So thank you for Pillar it definitely made me a more active contributor. Pillar has a long way to go to compete with Latex feature wise, but I think its on the right road.

Comment (DamienCassou) [inline Preface/Preface.pillar]: Indeed, Pillar can generate Markdown. But because Markdown has less features, the output will be less complete.

Comment (dmitrizagidulin) [inline Preface/Preface.pillar]: > When you prefix a line with @@XXX Pillar generates a LaTeX environment named XXX. It's up to the > book author to define this XXX LaTeX environment. This is typically in a file named support/latex/common.tex.

Ohh, I see! I thought it was hardcoded into Pillar itself. Ok, I'll fix the 'Notes' phrasing.
Is there a way to get these annotated tags to generate different HTML tags? Like, a way to specify that `@@quote` should generate an HTML `<blockquote>`?

Comment (dmitrizagidulin) [inline Preface/Preface.pillar]: > LaTeX is heavy to install so people tend not to contribute

It could be argued that (at least for these books), you can't really contribute in Pillar without installing LaTeX anyway :) 

(The issue is fresh in my mind cause I'm having various version issues at the moment and having to upgrade to LiveTeX 2015 :) ).

Since the PDF is the main product (along with the HTML), it's not very useful for people to contribute to the book in any significant way (aside from a minor spelling fix) without checking if their changes broke the generated PDF.

So, the difficulty of installation is unfortunately not removed with Pillar, but the main benefit of not having to know LaTeX syntax is pretty significant.

Comment (DamienCassou) [inline Preface/Preface.pillar]: Dmitri Zagidulin <notifications@github.com> writes:
> Is there a way to get these annotated tags to generate different HTML
> tags? Like, a way to specify that `@@quote` should generate an HTML
> `<blockquote>`?

that's unfortunately not possible

-- 
Damien Cassou
http://damiencassou.seasidehosting.st

"Success is the ability to go from one failure to another without
losing enthusiasm." --Winston Churchill

Comment (DamienCassou) [inline Preface/Preface.pillar]: kilon <notifications@github.com> writes:
> Pillar has a long way to go to compete with Latex feature wise

that's not the goal. I don't plan to make Pillar Turing-complete :-).

-- 
Damien Cassou
http://damiencassou.seasidehosting.st

"Success is the ability to go from one failure to another without
losing enthusiasm." --Winston Churchill

Comment (DamienCassou) [inline Preface/Preface.pillar]: zio-pietro <notifications@github.com> writes:
> @dmitrizagidulin  yes, I'm slowly discovering the background
> (personally I think it's a kind of NIH syndrom),


Most of Pillar's implementation and syntax (formerly known as Pier,
formerly known as Smallwiki, formerly known as Swiki) probably predates
all markups you know. So I can happily say that other markups have a
kind of NIH syndrome :-). But I agree that Pillar was (and is still) not
accessible to non-Smalltalk developers so it doesn't count.

> but I'll do my best of helping us to do good UPBE from typographics
> point of view even with Pillar. Still remain of opinion for short
> texts any markup language, for big works LaTeX. "Here I stand. I can
> do no other" [two hours long continuation deleted, I know, it's a lost
> battle, today's people want quick results with minimal effort and
> so-so quality is good enough]


I agree that LaTeX would allow best quality. But I think that
EnterprisePharo is already very good for something generated out of a
markup language. And we have HTML too.

-- 
Damien Cassou
http://damiencassou.seasidehosting.st

"Success is the ability to go from one failure to another without
losing enthusiasm." --Winston Churchill
